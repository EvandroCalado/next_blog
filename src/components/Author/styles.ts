import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
${({ theme }) => css`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 3rem;
  margin: ${theme.spacings.xxlarge} 0;
`}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 15rem;
    border: 3px solid ${theme.colors.secondary};
    border-radius: 50%;
  `}
`;

export const Container = styled.div`
`;

export const Title = styled.h2`
${({ theme }) => css`
font-size: ${theme.font.sizes.xlarge};
`}
`;

export const Paragraph = styled.p`
`;
