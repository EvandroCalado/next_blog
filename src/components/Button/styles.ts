import styled, { css } from 'styled-components';
import { ButtonProps } from './';

export const Button = styled.button<Pick<ButtonProps, 'color'>>`
  ${({ theme, color }) => css`
    background-color: ${theme.colors[color]};
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.medium};
    border: none;
    padding: ${theme.spacings.xsmall} ${theme.spacings.medium};
    margin: ${theme.spacings.large} 0;
    border-radius: 0.5rem;
    display: flex;
    align-items: center;
    justify-content: center;
    transition: all 300ms ease-in-out;
    cursor: pointer;

    &:hover {
      filter: brightness(80%);
    }

    &:disabled {
      background-color: ${theme.colors.mediumGray};
      color: ${theme.colors.darkGray};
      cursor: not-allowed;

      &:hover {
        filter: none;
      }
    }

    > svg {
      width: 3rem;
      margin-left: 1rem;
    }
  `}
`;
