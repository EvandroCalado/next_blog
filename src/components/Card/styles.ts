import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
    a {
      transition: all 300ms ease-in-out;
    }

    a:hover {
      opacity: 1;
      filter: brightness(170%);
    }
`;

export const Cover = styled.div`
  ${({ theme }) => css`
    img {
      width: 100%;
      display: block;
      aspect-ratio: 16/9;
      border-radius: .5rem;
      margin-bottom: ${theme.spacings.small};

    }
  `}
`;

export const Heading = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.smallMedium};
    color: ${theme.colors.darkGray};
    margin-bottom: ${theme.spacings.xsmall};
  `}
`;

export const Paragraph = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    margin-bottom: ${theme.spacings.small};
    color: ${theme.colors.primary};
  `}
`;
