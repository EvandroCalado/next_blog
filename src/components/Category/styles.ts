import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
${({ theme }) => css`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 2rem;
  margin: ${theme.spacings.xxlarge} 0;
`}
`;

export const Image = styled.img`
  width: 8rem;
  border-radius: 0.3rem;
`;

export const Title = styled.h2`
${({ theme }) => css`
  font-size: ${theme.font.sizes.xlarge};

  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.large};
  }
`}
`;
