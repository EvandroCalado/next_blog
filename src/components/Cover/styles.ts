import styled, { css } from 'styled-components';

export const Image = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    aspect-ratio: 16/9;
    object-fit: cover;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
