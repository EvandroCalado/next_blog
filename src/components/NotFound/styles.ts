import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    color: ${theme.colors.primary};
    font-size: ${theme.font.sizes.xxlarge};
    width: 100%;
    height: 60rem;
    margin: ${theme.spacings.xlarge} 0;
  `}
`;
