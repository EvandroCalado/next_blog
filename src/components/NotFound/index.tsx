import Heading from '../Heading';
import * as Styled from './styles';

export type NotFoundProps = {
  children: React.ReactNode;
};

const NotFound = ({ children = '' }: NotFoundProps) => {
  return (
    <Styled.Wrapper>
      <Heading>{children}</Heading>
    </Styled.Wrapper>
  );
};

export default NotFound;
