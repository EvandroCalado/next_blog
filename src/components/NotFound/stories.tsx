import NotFound, { NotFoundProps } from './';
import { Meta, StoryFn } from '@storybook/react';

export default {
  title: 'NotFound',
  component: NotFound,
  args: {
    children: 'NÃ£o encotrado',
  },
} as Meta;

export const Template: StoryFn<NotFoundProps> = (args) => (
  <div>
    <NotFound {...args} />
  </div>
);
