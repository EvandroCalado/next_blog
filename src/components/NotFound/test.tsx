import NotFound from './';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';

describe('<NotFound />', () => {
  it('test_children_prop_is_rendered_correctly', () => {
    renderTheme(<NotFound>test</NotFound>);
    expect(screen.getByText('test')).toBeInTheDocument();
  });

  it('test_renders_not_found_component_with_children_prop', () => {
    renderTheme(<NotFound>{''}</NotFound>);
    expect(screen.queryByText('test')).not.toBeInTheDocument();
  });

  it('test_render_without_children_prop', () => {
    renderTheme(<NotFound>{}</NotFound>);
    expect(screen.getByRole('heading')).toBeInTheDocument();
  });

  it('test_render_snapshop', () => {
    const { container } = renderTheme(<NotFound>test</NotFound>);
    expect(container).toMatchSnapshot();
  });
});
