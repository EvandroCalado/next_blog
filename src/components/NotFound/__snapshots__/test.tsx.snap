// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> test_render_snapshop 1`] = `
.c1 {
  font-size: 5.6rem;
  margin: 3.2rem 0;
  text-align: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #252422;
  font-size: 4.8rem;
  width: 100%;
  height: 60rem;
  margin: 4.0rem 0;
}

@media (max-width:768px) {
  .c1 {
    font-size: 3.2rem;
  }
}

<div>
  <div
    class="c0"
  >
    <h2
      class="c1"
      data-testid="heading"
    >
      test
    </h2>
  </div>
</div>
`;
